src/Microsoft.NetCore.Analyzers/Core/Security/DoNotOverloadDeserializationBinderWithoutThrowingAnException.cs

```diff
@@ -0,0 +1,158 @@
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(
+                (CompilationStartAnalysisContext compilationStartAnalysisContext) =>
+                {
+                    Dictionary<IMethodSymbol, CacheState> methodCache = new Dictionary<IMethodSymbol, CacheState>();
+                    WellKnownTypeProvider typeProvider = WellKnownTypeProvider.GetOrCreate(compilationStartAnalysisContext.Compilation);
+                    INamedTypeSymbol SerializationBinder = null;
+                    if (!typeProvider.TryGetTypeByMetadataName("System.Runtime.Serialization.SerializationBinder", out SerializationBinder))
+                    {
+                        System.Diagnostics.Debug.Fail("Unable to find System.Runtime.Serialization.SerializationBinder threw WellKnownTypeProvider");
+                        return;
+                    }
+
+                    compilationStartAnalysisContext.RegisterSymbolAction(
+                        (SymbolAnalysisContext symbolAnalysisContext) =>
+                        {
+                            IMethodSymbol msym = (IMethodSymbol)symbolAnalysisContext.Symbol;
+                            if (msym.Name == "BindToType")
+                            {
+                                if (ITypeSymbolExtensions.Inherits(msym.ContainingType, SerializationBinder))
+                                {
+                                    IBlockOperation msymOps = msym.GetTopmostOperationBlock(compilationStartAnalysisContext.Compilation);
+                                    methodCache[msym] = CacheState.InProgress;
+                                    if (!hasThrow(msymOps.Operations))
+                                    {
+                                        symbolAnalysisContext.ReportDiagnostic(
+                                            Diagnostic.Create(
+                                                DoNotOverloadSerializationBinderWithoutThrowingAnExceptionRule,
+                                                msym.Locations[0],
+                                                msym.ContainingType.ToDisplayString(
+                                                    SymbolDisplayFormat.MinimallyQualifiedFormat)));
+                                    }
+                                }
+                            }
+                        }
+                    , SymbolKind.Method);
+
+                    bool hasThrow(IEnumerable<Microsoft.CodeAnalysis.IOperation> ops)
+                    {
+                        if (ops == null)
+                        {
+                            //doesn't happen, but just in case
+                            System.Diagnostics.Debug.Fail("hasThrow called on null value. It should be called on an IEnumerable of IOperations.");
```

Comment:
>System.Diagnostics. [](start = 28, length = 19)

Nit: Add a using for System.Diagnostics so you don't have to specify the full namespace here.